client/blocks/post-share/publicize-actions-list.jsx

```diff
@@ -41,14 +42,29 @@ class PublicizeActionsList extends PureComponent {
 	state = {
 		selectedShareTab: SCHEDULED,
 		showDeleteDialog: false,
-		selectedScheduledShareId: null
+		selectedScheduledShareId: null,
+		showPreviewModal: false,
+		previewMessage: '',
+		previewService: '',
 	};
 
 	setFooterSection = selectedShareTab => () => {
 		analytics.tracks.recordEvent( 'calypso_publicize_action_tab_click', { tab: selectedShareTab } );
 		this.setState( { selectedShareTab } );
 	};
 
+	togglePreviewModal = ( message = '', service = '' ) => () => {
+		if ( this.state.showPreviewModal ) {
+			this.setState( { showPreviewModal: false } );
+		} else {
```

Comment:
minor suggestion.

In general, I prefer to avoid the `else` here using a `return` into the `if`.

```es6
togglePreviewModal = ( message = '', service = '' ) => () => {
 	if ( this.state.showPreviewModal ) {
 		return this.setState( { showPreviewModal: false } );
 	}

	this.setState( {
		showPreviewModal: true,
		previewMessage: message,
		previewService: service,
	} );
}
```
Less tabbing, less code. Please feel free to ignore it.